### PR TITLE
Bump Cirrus image Xcode version

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Update Cirrus macOS image to catalina-xcode-12.0-flutter.

## Related Issues

https://github.com/flutter/flutter/issues/60133

### CODE REVIEW

**Comment by fkorotkov** [comment]
@jmagman you can re-run the macOS tasks. `catalina-xcode-12.0-flutter` image is available now.

**Comment by jmagman** [comment]
> @jmagman you can re-run the macOS tasks. `catalina-xcode-12.0-flutter` image is available now.

Kicked it off, thank you!

**Comment by jmagman** [comment]
Investigating new test failure:
```
[plugin_lint_mac] [STDOUT] stdout:     ld: building for iOS Simulator, but linking in dylib built for iOS, file '/var/folders/3y/l0z1x3693dl_8n0qybp4dqwh0000gn/T/CocoaPods-Lint-20200812-67637-y2vglb-test_plugin_objc/Pods/Flutter/Flutter.framework/Flutter' for architecture arm64
[plugin_lint_mac] [STDOUT] stdout:     clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by xster** [comment]
LGTM

**Comment by fkorotkov** [comment]
@jmagman since this PR is not yet merged, will it be OK if I'll update he 12.0 image to have Android SDK 30? Seem #63517 will be merged first?

**Comment by jmagman** [comment]
> @jmagman since this PR is not yet merged, will it be OK if I'll update he 12.0 image to have Android SDK 30? Seem #63517 will be merged first?

@fkorotkov Sure, I haven't had time to investigate the new test failure.

**Comment by jmagman** [comment]
@xster That's a known warning, the error is that the `ld: building for iOS Simulator, but linking in dylib built for iOS`

**Comment by jmagman** [comment]
Looks like it created an arm simulator destination instead of an x86_64 one, and Flutter only has a x86_64 simulator slice (the arm slice is iphoneos not iphonesimulator).
arm macOS developers will see the same error when they try to run Flutter on simulators...

**Comment by jmagman** [comment]
Filed https://github.com/flutter/flutter/issues/64502.

File: dev/devicelab/bin/tasks/plugin_lint_mac.dart [hunk 1]
```diff
@@ -50,6 +50,8 @@ Future<void> main() async {
           environment: <String, String>{
             'LANG': 'en_US.UTF-8',
           },
+          // TODO(jmagman): Flutter cannot build against ARM simulators https://github.com/flutter/flutter/issues/64502
+          canFail: true,
```
Comment (jmagman) [inline dev/devicelab/bin/tasks/plugin_lint_mac.dart:54]: This sucks.
The plugin linter tests are going to hit this problem with Xcode 12, too.
\cc @cyanglaz

**Comment by xster** [comment]
LGTM

File: dev/devicelab/bin/tasks/plugin_lint_mac.dart (hunk 1 above)
Comment (cyanglaz) [reply dev/devicelab/bin/tasks/plugin_lint_mac.dart:54]: I'm not familiar with the syntax. Doesn "canFail" make the analyzer pass even hitting this problem?

Comment (jmagman) [reply dev/devicelab/bin/tasks/plugin_lint_mac.dart:54]: Yes, it's running this check but not failing the test if the linter fails.  Flutter is missing an ARM simulator slice, so will fail on Xcode 12.

**Pull request merged**